src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -281,10 +282,10 @@ private static string GenerateUpgradeReport(string projectName, string backupPat
             }
         }
 
-        private static Tuple<string, string> CreatePackageTuple(IPackageReferenceContextInfo pkg)
+        private static Tuple<string, string, string> CreatePackageTuple(IPackageReferenceContextInfo pkg)
         {
             PackageIdentity package = pkg.Identity;
-            return Tuple.Create(package.Id, package.Version == null ? string.Empty : package.Version.ToNormalizedString());
+            return Tuple.Create(package.Id, package.Version == null ? string.Empty : package.Version.ToNormalizedString(), pkg.AllowedVersions.OriginalString);
```

Comment:
AllowedVersions and OriginalString can be nulls, I added a null check for that

Reply:
Yes we send the same information for packages.config and for that case versionRange is going to be the same as version, its like installing a non floating version. Should we avoid sending that telemetry when the project is not PR?

Reply:
As long as we use/interpret the data in a correct way, then it probably doesn't matter. 
Ideally we have the project type in the same event.

Reply:
I was wrong, this function will be used for existingPackages and for packages.config projects we dont have AllowedVersions property, so versionRange will not be send in the Telemetry. Also, we dont have the project type in the same event

![image](https://user-images.githubusercontent.com/43253759/220775924-46e8ef3e-14c7-4ed6-8065-eb8bd0fda279.png)

Reply:
I think the project type would be helpful in either case. 


> for packages.config projects we dont have AllowedVersions property, so versionRange will not be send in the Telemetry

The original concept actually comes form packages.config

Test out:
```
<packages>
  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net45" allowedVersions="[6.0.0, 10.0.0)" />
</packages>
```

Reply:
You're right, it will be added to telemetry as `{"version":"8.0.2","versionRange":"[6.0.0, 10.0.0)","id":"newtonsoft.json"}`

Should I try to add the projectType in this PR or follow up in a new one?

Reply:
Whatever you prefer. I don't mind having different PRs.

Reply:
Just let us know what you picked so that we know when we should review again :)

Reply:
Created bug to follow up https://github.com/NuGet/Client.Engineering/issues/2166